### PR TITLE
Wire up transactional emails and a rudimentary django-admin CMS for managing them 

### PR DESCRIPTION
**Pull request by ejucovy**
* If a given email `type` isn't defined, the send_email function does nothing
* If multiple emails are defined for a given `type`, select one randomly to send -- so that administrators can A/B test different email subjects & bodies  

Emails that can currently be wired up are:
* When a moderator removes an inappropriate idea, removes a duplicate idea, or merges two ideas
* When a user submits a new idea

This will probably have performance implications unless we wire up `django-celery-email` or something.

I know the `order_by("?")` is inefficient in theory, but I'm assuming it won't matter in practice since any given `type` would have 0-5 versions wired up at most.

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ django-localflavor==1.2
 django-endless-pagination==2.0
 django-dotenv==1.4.1
 djangohelpers==0.18.1
-djorm-ext-pgfulltext==0.9.3
+git+git://github.com/linuxlewis/djorm-ext-pgfulltext@tags/0.9.3
```
Comment (ejucovy) [inline requirements/base.txt]: Shoot, sorry -- didn't mean to include this.  (I wasn't able to find this requirement unless I updated it to point directly at the github version.  But still didn't intend to put it in this commit...)

Comment (dpoirier) [inline requirements/base.txt]: You can update the branch to undo this change.

**Comment by dpoirier** [comment]
I agree we'll want to send these emails in the background, but we can add that after we get the basic send function working.

Can we add a test or two?  Maybe create an EmailTemplate, then submit an idea and check that Django's test mailbox got an email and its contents look right?  That's more of an integration test than a unit test, but given our time constraints, I think it'll be good enough for now.

File: opendebates_emails/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+#
```
Comment (dpoirier) [inline opendebates_emails/__init__.py]: ?

**Comment by dpoirier** [comment]
@ejucovy please leave a comment when this is ready to be reviewed again. Unfortunately Github doesn't notify anyone when a pull request's code is updated.

File: opendebates_emails/__init__.py (hunk 2 above)
Comment (ejucovy) [inline opendebates_emails/__init__.py]: Is there a better way to add an empty file into version control?  This is how I've always done it (a blank file with just an empty comment)

**Comment by ejucovy** [comment]
Ah, sorry.  This should be ready for review.  I've added some basic integration tests to verify that (a) an email will be sent when submitting an idea if the appropriate `type` of `EmailTemplate` has been set up, (b) that the email's subject, html and text body, to and from addresses are all rendered from templates with `ctx={"idea": new_submission}`, and (c) some rudimentary explanation of what happens when multiple EmailTemplates are created of the same `type`.

File: opendebates_emails/__init__.py (hunk 2 above)
Comment (dpoirier) [inline opendebates_emails/__init__.py]: A plain old empty file works fine.

File: opendebates/moderator_views.py [hunk 3]
```diff
@@ -46,15 +47,12 @@ def mark_duplicate(request):
         votes_to_merge.update(original_merged_submission=to_remove, submission=duplicate_of)
 
     if request.POST.get("send_email") == "yes":
-        # FIXME: `emails` was not defined, so I have commented it out for now. --vkurup
         if request.POST.get("handling") == "merge":
-            pass
-            # emails.idea_is_merged(to_remove)
+            send_email("your_idea_is_merged", {"idea": to_remove})
```
Comment (dpoirier) [inline opendebates/moderator_views.py]: It seems like all of these need to pass in the idea submitter's email address so send_email will know where to send the mail to.

Comment (ejucovy) [inline opendebates/moderator_views.py]: See tests -- I set up the recipient field (and all other info that's needed to send the mail) as a templated attribute on the `EmailTemplate` object.  So the `EmailTemplate` object plus the idea object can fully determine the email behavior.  The admins would use the Django Admin Site TTW to create an `EmailTemplate` with `recipient = {{ idea.voter.email }}`.
